### PR TITLE
Add Jenkinsfile Runner

### PR DESCRIPTION
**Pull request by fwilhe**

### CODE REVIEW

File: jenkinsfile-runner/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM maven:3.5.2 as maven
+
+RUN git clone https://github.com/jenkinsci/jenkinsfile-runner && \
```
Comment (daniel-kurzynski) [inline jenkinsfile-runner/Dockerfile]: The Dockerfile for the jenkinsfile-runner uses a multi stage build to cache the mvn dependencies.

File: jenkinsfile-runner/README.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+## Build
+
+To build this image locally, open a terminal in the directory of the Dockerfile an run
+
+```
+docker build -t ppiper/jenkinsfile-runner .
+```
+
+**Warning:** Building this image will take quite long, as the Jenkinsfile Runner is built from source and has many dependencies.
+Expect high network traffic.
```
Comment (daniel-kurzynski) [inline jenkinsfile-runner/README.md]: See comment above

Comment (fwilhe) [reply jenkinsfile-runner/README.md]: Regardless of cache, the first build will always take long and download lots of dependencies. Also, Docker does cache the layer, so I'm not really sure if I understand the benefit of what they do in the upstream Dockerfile. I'd wanted to keep it simple. What is the benefit of what they do?

**Comment by fwilhe** [comment]
Hello everyone, this is a prerequisite for the infrastructure testing I'm working on, so a timely review is highly appreciated. If I need to provide more context, please tell me. Thanks :)

**Pull request merged**